pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -978,6 +986,75 @@ func isGenericCSIPersistentVolumeClaim(pvc *v1.PersistentVolumeClaim, volInfos [
 	return false, nil
 }
 
+func (a *ApplicationRestoreController) getPVCFromObjects(objects []runtime.Unstructured, volumeBackupInfo *storkapi.ApplicationBackupVolumeInfo) (*v1.PersistentVolumeClaim, error) {
+	var pvc v1.PersistentVolumeClaim
+	for _, o := range objects {
+		objectType, err := meta.TypeAccessor(o)
+		if err != nil {
+			return &pvc, err
+		}
+
+		if objectType.GetKind() == "PersistentVolumeClaim" {
+			var tempPVC v1.PersistentVolumeClaim
+			err := runtime.DefaultUnstructuredConverter.FromUnstructured(o.UnstructuredContent(), &tempPVC)
+			if err != nil {
+				return &pvc, nil
+			}
+			if tempPVC.Name == volumeBackupInfo.PersistentVolumeClaim {
+				pvc = tempPVC
+				break
+			}
+		}
+	}
+	return &pvc, nil
+}
+
+func (a *ApplicationRestoreController) skipVolumesFromRestoreList(
+	backup *storkapi.ApplicationBackup,
+	restore *storkapi.ApplicationRestore,
+	objects []runtime.Unstructured,
+	driver volume.Driver,
+) ([]*storkapi.ApplicationBackupVolumeInfo, []*storkapi.ApplicationRestoreVolumeInfo) {
+	existingInfos := make([]*storkapi.ApplicationRestoreVolumeInfo, 0)
+	newVolInfos := make([]*storkapi.ApplicationBackupVolumeInfo, 0)
+	for _, bkupVolInfo := range backup.Status.Volumes {
+		restoreVolInfo := &storkapi.ApplicationRestoreVolumeInfo{}
+
+		val, ok := restore.Spec.NamespaceMapping[bkupVolInfo.Namespace]
+		if !ok {
+			newVolInfos = append(newVolInfos, bkupVolInfo)
+			continue
+		}
+
+		// get corresponding pvc object from objects list
+		pvcObject, err := a.getPVCFromObjects(objects, bkupVolInfo)
+		if err != nil {
+			newVolInfos = append(newVolInfos, bkupVolInfo)
+			continue
+		}
+
+		ns := val
+		pvc, err := core.Instance().GetPersistentVolumeClaim(pvcObject.Name, ns)
+		if err != nil {
+			newVolInfos = append(newVolInfos, bkupVolInfo)
```

Comment:
In the case of errors, should we really return error. We check for notfound error and it is notfound error, we can continue.

Reply:
in case of error while restoring pvcs, we are continuing even if one of them fails and marking the whole restore as failed. Similarly here the idea is to create two different lists (one with already existing and one where restore should really take place) and if down the line, restore hits any error it will have the status as Failed for that volume.

Reply:
addressed the comment.